### ISSUE TITLE
Redis Extension Bug - Upgrade needed

### ISSUE THREAD

**Issue by wapacro** [cloudfoundry/php-buildpack#322]
_What version of the buildpack you are using?_
v4.3.79 - v4.3.82

_If you were attempting to accomplish a task, what was it you were attempting to do?_
Using the included Redis extension to connect to a Redis Cluster

_What did you expect to happen?_
Successful connection to Redis

_What was the actual behavior?_
Exception thrown stating "No such file or directory"
```
No such file or directory {"exception":"[object] (RedisException(code: 0): No such file or directory at /home/vcap/app/vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php:120)
   2019-09-11T18:31:14.92+0200 [APP/PROC/WEB/0] OUT 16:31:14 nginx   | [stacktrace]
   2019-09-11T18:31:14.92+0200 [APP/PROC/WEB/0] OUT 16:31:14 nginx   | #0 /home/vcap/app/vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php(120): Redis->connect('kubernetes-serv...', 49948, 0, NULL, 0, 0)
   2019-09-11T18:31:14.92+0200 [APP/PROC/WEB/0] OUT 16:31:14 nginx   | #1 /home/vcap/app/vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php(77): Illuminate\\Redis\\Connectors\\PhpRedisConnector->establishConnection(Object(Redis), Array)
   2019-09-11T18:31:14.92+0200 [APP/PROC/WEB/0] OUT 16:31:14 nginx   | #2 /home/vcap/app/vendor/laravel/framework/src/Illuminate/Support/helpers.php(424): Illuminate\\Redis\\Connectors\\PhpRedisConnector->Illuminate\\Redis\\Connectors\\{closure}(Object(Redis))
   2019-09-11T18:31:14.92+0200 [APP/PROC/WEB/0] OUT 16:31:14 nginx   | #3 /home/vcap/app/vendor/laravel/framework/src/Illuminate/Redis/Connectors/PhpRedisConnector.php(94): tap(Object(Redis), Object(Closure))
```

_Reproduction steps:_
- Setup a Redis connection using the PhpRedis (`redis.so`) extension using a port > 32768

_Reason:_
With the release of PHP Buildpack v4.3.79 the PhpRedis extension was upgraded to v5.0.1. But this release of the extension contains a bug which prevents connecting to Redis using a port greater than 32768. See https://github.com/phpredis/phpredis/pull/1602.

Our CF provider uses Redis port 49948 which means we have to stick with the buildpack version v4.3.78 and cannot benefit from PHP 7.3.9 and the various improvements of PhpRedis.

_Recommendation:_
Upgrade PhpRedis extension to v5.0.2 to fix this. Additionally this will make the extension ready for PHP 7.4.

_Please confirm where necessary:_
* [x] I have included a log output
* [X] My log includes an error message
* [X] I have included steps for reproduction

**Comment by ocean90** [comment]
If you don't depend on the PhpRedis 5.0 you can write a custom extension which replaces `redis.so` with the version from v4.3.78:

* Download the PHP package used in v4.3.78:
https://buildpacks.cloudfoundry.org/dependencies/php/php7-7.3.6-linux-x64-cflinuxfs3-1a76a4dd.tgz
* Extract `redis.so` from `php7-7.3.6-linux-x64-cflinuxfs3-1a76a4dd/php/lib/php/extensions/no-debug-non-zts-20180731/redis.so`
* Copy `redis.so` into `.extensions/php-redis/`
* Create `extension.py` with the content from below
<details>
<summary>.extensions/php-redis/extension.py</summary>

```python
"""Replace PHP redis module

Replaces the redis.so with the one from php7-7.3.6-linux-x64-cflinuxfs3-1a76a4dd.
See https://github.com/cloudfoundry/php-buildpack/issues/322.
"""
import os
import os.path
import logging
from build_pack_utils import utils
import shutil


_log = logging.getLogger('php-redis')

# Extension Methods
def preprocess_commands(ctx):
    return ()

def service_commands(ctx):
    return {}

def service_environment(ctx):
    return {}

def compile(install):
    print 'Replacing PHP redis module'

    ctx = install.builder._ctx
    dest   = os.path.join(ctx['BUILD_DIR'], 'php', 'lib', 'php', 'extensions', 'no-debug-non-zts-20180731', 'redis.so')
    source = os.path.join(ctx['BUILD_DIR'], '.extensions', 'php-redis', 'redis.so')
    shutil.copy(source, dest)

    return 0
```

</details>

**Comment by Zanadar** [comment]
Will technically be delivered when the rebuilds end up in the buildpack.

**Comment by ocean90** [comment]
This can be closed, [4.4.0 with the fix has been released](https://github.com/cloudfoundry/php-buildpack/releases/tag/v4.4.0). ✨
